test/compress/destructuring.js

```diff
@@ -200,3 +200,13 @@ destructuring_remove_unused_2: {
         }
     }
 }
+
+object_destructuring_may_need_parentheses: {
+    beautify = {
+        ecma: 6
+    }
+    input: {
+        ({a, b} = {a: 1, b: 2});
```

Comment:
I'm not going to remove this behaviour. But remembering the spec, it should be that the sum of the tokens should be a valid lhs-expression. Might be better to watch out for side-effects or non-intended behaviour though (but that isn't our first priority for now).

Not removing, since we have/had a lot non-standard behaviour, but recommending users to stick to the standard rules in case there are behaviour changes.